vhdbuilder/packer/generate-vhd-publishing-info.sh

```diff
@@ -81,23 +80,29 @@ else
     IMAGE_ARCH="x64"
 fi
 
-echo "generating traditional SAS token with CLASSIC_SA_CONNECTION_STRING..."
-start_date=$(date +"%Y-%m-%dT00:00Z" -d "-1 day")
-expiry_date=$(date +"%Y-%m-%dT00:00Z" -d "+1 year")
 if [[ "${OS_NAME,,}" != "windows" ]]; then
-    [ -z "${OUTPUT_STORAGE_CONTAINER_NAME}" ] && echo "OUTPUT_STORAGE_CONTAINER_NAME should be set..." && exit 1
+    # for Linux we use the managed identity attached to the 1ES pool agents to generate SAS tokens without a connection string
+    echo "linux VHD - generating user-delegation SAS token for publishing-info creation"
+    [ -z "${OUTPUT_STORAGE_ACCOUNT_NAME}" ] && echo "OUTPUT_STORAGE_ACCOUNT_NAME should be set when generating user-delegation SAS token with agent identity" && exit 1
+    [ -z "${OUTPUT_STORAGE_CONTAINER_NAME}" ] && echo "OUTPUT_STORAGE_CONTAINER_NAME should be set when generating user-delegation SAS token with agent identity" && exit 1
+    echo "storage account name: ${OUTPUT_STORAGE_ACCOUNT_NAME}"
     echo "storage container name: ${OUTPUT_STORAGE_CONTAINER_NAME}"
     # max of 7 day expiration time when using user delegation SAS
-    sas_token=$(az storage container generate-sas --name ${OUTPUT_STORAGE_CONTAINER_NAME} --permissions lr --connection-string ${CLASSIC_SA_CONNECTION_STRING} --start ${start_date} --expiry ${expiry_date} | tr -d '"')
+    expiry_date=$(date +"%Y-%m-%dT00:00Z" -d "+7 day")
```

Comment:
This introduces a requirement. The new image must be published to all cloud in 7 days. It will fail if it is stuck in some regions/clouds for some reasons.

Reply:
Do you think whether it is ok to generate user-delegation SAS tokens in creating artifacts or publishing the images?